test/firebase_auth_mocks_test.dart

```diff
@@ -19,17 +19,18 @@ void main() {
       final user = await result.user;
       expect(user.uid, isNotEmpty);
       expect(user.displayName, isNotEmpty);
-      expect(auth.onAuthStateChanged, emitsInOrder([user]));
+      expect(auth.onAuthStateChanged, emitsInOrder([isA<User>()]));
```

Comment:
I made the user getter return a new instance every time, so the previous equality test would fail.